lib.rs

```diff
@@ -479,12 +505,30 @@ impl Drop for AsyncCore {
 }
 // }}}
 
+/// Behavior used when the channel is full.
+#[derive(Copy, Clone, Debug, Eq, PartialEq, Hash, Ord, PartialOrd)]
+pub enum OverflowStrategy {
+    /// The message gets dropped and a message with number of dropped is produced once there's
+    /// space.
+    ///
+    /// This is the default.
+    ///
+    /// Note that the message with number of dropped messages takes one slot in the channel as
+    /// well.
+    DropAndReport,
+    /// The message gets dropped silently.
+    Drop,
+    /// The caller is blocked until there's enough space.
+    Block,
+}
+
 /// `Async` builder
 pub struct AsyncBuilder<D>
 where
     D: slog::Drain<Err = slog::Never, Ok = ()> + Send + 'static,
 {
     core: AsyncCoreBuilder<D>,
+    inc_dropped: bool,
```

Comment:
I don't immediately know what it means, so it probably should be a different name.

Please also add a comment.